index.html

```diff
@@ -827,8 +827,36 @@ <h3>
       </li>
       <li>
         <p>
-          Instruct the [= ICE agent =] to use |candidatePair| to send data, and continue to the steps for a change in the selected
-          candidate pair (leading up to {{RTCIceTransport/onselectedcandidatepairchange}}).
+          Let |p:Promise| be a new promise.
+        </p>
+      </li>
+      <li>
+        <p>
+          In parallel, instruct the [= ICE agent =] to use |candidatePair| to send data.
+        </p>
+        <ol>
+          <li>
+            <p>
+              When the [= ICE agent =] has completed selecting |candidatePair|, [= queue a task =] to run the following steps:
```

Comment:
We need to resolve how this interacts with [§ 5.6 RTCIceTransport Interface](https://www.w3.org/TR/webrtc/#rtcicetransport)'s _"When the [ICE Agent](https://www.w3.org/TR/webrtc/#dfn-ice-agent) indicates that an [RTCIceTransport](https://www.w3.org/TR/webrtc/#dom-rtcicetransport) has changed either the selected candidate pair, the [RTCIceTransportState](https://www.w3.org/TR/webrtc/#dom-rtcicetransportstate) or both, the user agent MUST queue a task that runs the following steps"_

That one task fires `selectedcandidatepairchange`, `statechange`, `iceconnectionstatechange` and `connectionstatechange` (or a subset) all on the same task, in that order.

Since that task goes to surface all that state at once

Options I see:
1. Fire the events one task _after_ the promise resolves (what this PR does I believe)
2. Fire the events one task _ahead_ of the promise resolving (would take some finagling)
3. Fire the events on the _same_ task _ahead_ of the promise resolving

Given the lengths webrtc-pc already goes to fire all 4 events on the same task, I suggest 3. This might also be the simplest: Do a PR on webrtc-pc to name the algorithm and call it synchronously from the queued task here which also resolves the promise.

Reply:
3 was my intention as well. PR on webrtc-pc sounds good.

Reply:
I am also in favor of 3. But shall we do that as a separate isse and PR? Since it requires following up on webrtc-pc and since this part of steps are not new to this PR.

@sam-vi Will you file the appropriate webrtc-pc/webrtc-extensions issues?

Reply:
w3c/webrtc-pc#2904 and #185 filed.

Reply:
Thank you!